x-pack/elastic-agent/pkg/composable/providers/kubernetes/config.go

```diff
@@ -25,6 +27,16 @@ type Config struct {
 
 	// Needed when resource is a Pod or Node
 	Node string `config:"node"`
+
+	AddResourceMetadata *metadata.AddResourceMetadataConfig `config:"add_resource_metadata"`
+	IncludeLabels       []string                            `config:"include_labels"`
+	ExcludeLabels       []string                            `config:"exclude_labels"`
+
+	LabelsDedot      bool `config:"labels.dedot"`
+	AnnotationsDedot bool `config:"annotations.dedot"`
+
+	// Undocumented settings, to be deprecated in favor of `drop_fields` processor:
+	IncludeCreatorMetadata bool `config:"include_creator_metadata"`
```

Comment:
If undocumented and deprecated, why add it? Being this is all new to Elastic Agent, it could be the time to remove it.

Reply:
You are right this one is not exposed so it should be removed.

Reply:
Actually this touches code that is used by beats too. I will remove it in follow up PR target 8.0

Reply:
PR: https://github.com/elastic/beats/pull/28006